### PR TITLE
Adding improved documentation for main/deepnest.js...

### PR DESCRIPTION
**Pull request by qaim-b**
1. Addressed Issues: Complexity in Nesting Logic: The nesting process lacked sufficient documentation, making it difficult for new contributors to understand how the code worked. Unclear Function Purpose: Several functions had ambiguous names, leading to confusion during code review.

2. Reengineered Components: Documented key classes, methods, and important operations in the deepnest.js file. The added documentation explains the functionality of methods like importsvg, renderPolygon, and simplifyPolygon, making the code much more accessible. Renamed some internal variables to improve clarity (self to deepNestInstance where appropriate).

3. Reengineering Strategy: Used an enhancement strategy focused on improving documentation, naming conventions, and the overall code readability. This approach ensured that the existing functionality was not altered while making the code easier to understand and extend.

4. Impact of Changes: Improved codebase clarity, making it more welcoming for new contributors. Reduced the learning curve for developers and increased maintainability without altering existing functionality.

### CODE REVIEW

**Comment by suhadaudd11** [comment]
Well done @qaim-b, However im expecting you to use and relate with the reengineering strategy that we have learned

File: main/deepnest.js [hunk 1]
```diff
@@ -945,7 +956,7 @@
 		
 		// progressCallback is called when progress is made
 		// displayCallback is called when a new placement has been made
-		this.start = function(p, d){						
+		for = function(p, d){
```
Comment (Dexus) [inline main/deepnest.js:959]: This is crap!

**Comment by Dexus** [comment]
@suhadaudd11 @qaim-b  why do you no code review? why there are so stupid fails?

**Comment by suhadaudd11** [comment]
Dear @Dexus and team,

Sorry for the confusion and for not letting you know earlier. Just to let you know, this forked repository is now being used by my students for their assignment as they are learning and practicing software maintenance and evolution. Really appreciate your understanding in this matter and I hope this did not disrupt anything on your end

**Pull request merged**